### PR TITLE
Support multi-line export hoisting

### LINKED ISSUE #536: Transform fails when the opening bracket of an interface is not on the same line as the `interface` keyword
### What version of `astro` are you using?

1.1.3

### Are you using an SSR adapter? If so, which one?

None

### What package manager are you using?

npm

### What operating system are you using?

Linux

### Describe the Bug

I encountered this bug while declaring a `Props` interface that extended multiple interfaces. Because their names were long, prettier formatted it so the interface declaration spanned multiple lines, kind of like this:
```ts
interface Props
    extends Interface1, Interface2 {
    // ...interface properties
}
```

But this caused the following error:

> Transform failed with 1 error:
/home/projects/github-2nzrgw/src/components/Card.astro:29:0: ERROR: Expected "{" but found "const"

I thought it had to do with `extends` or the fact that I was using generics in my code, but after playing with it for a bit in StackBlitz I realized that it has to do with the opening bracket (`{`) not being on the same line as the `interface` keyword. 

```ts
// if we do this transform fails
interface Props extends Interface1, Interface2 
{
    // ...interface properties
}

// but if we do this then everything works fine
interface Props extends Interface1, Interface2 {
    // ...interface properties
}
```

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-2nzrgw?file=src/components/Card.astro

### Participation

- [ ] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by natemoo-re**
## Changes

- Closes #536
- Handles edge case with multi-line export hoisting
- Ensures types and interfaces are handled similarly to functions

## Testing

Test added

## Docs

Bug fix only

### CODE REVIEW

File: internal/js_scanner/js_scanner.go [hunk 1]
```diff
@@ -86,14 +86,18 @@ outer:
 					if isKeyword(nextValue) && next != js.FromToken {
 						continue
 					}
+					// ensure type declarations are defined or exported from package
+					if flags["type"] && !(flags["="] || flags["from"]) {
```
Comment (matthewp) [inline internal/js_scanner/js_scanner.go:90]: what is the `from` for?

Comment (natemoo-re) [reply internal/js_scanner/js_scanner.go:90]: This is valid TypeScript, apparently
```
export type Value from "package"
```

**Review by matthewp — APPROVED** [review]
